src/dataworker/Dataworker.ts

```diff
@@ -2719,6 +2724,80 @@ export class Dataworker {
     return l2MessagerContract.minimumFeeInWei();
   }
 
+  /**
+   * Compute the msg.value to attach when executing a relayer refund leaf on L2.
+   * - If amountToReturn == 0 or non-EVM chain: undefined
+   * - If OFT is configured on the SpokePool for the token: quote native fee via OFT messenger and return fee*2
+   * - Else if Linea: return the pre-fetched Linea L2->L1 message fee
+   * - Additionally, enforce OFT-on-Linea mutual exclusion and throw if encountered
+   */
+  private async _getMsgValueForRelayerRefundLeaf(
+    client: SpokePoolClient,
+    leaf: RelayerRefundLeaf
+  ): Promise<BigNumber | undefined> {
+    // We don't support providing SOL value with the executor call, so if spokePool is not evm, or the return amount is 0, exit early
+    if (!isEVMSpokePoolClient(client) || !leaf.amountToReturn.gt(0)) {
+      return undefined;
+    }
+
+    // If the chain is Linea, then we need to allocate ETH in the call to executeRelayerRefundLeaf
+    const lineaMsgValuePortion = sdkUtils.chainIsLinea(client.chainId)
+      ? await this._getRequiredEthForLineaRelayLeafExecution(client)
+      : BigNumber.from(0);
+
+    // If the SpokePool supports withdrawing tokens to Hub via OFT, estimate msg.value needed to cover OFT fee
+    const oftMsgValuePortion = await this._getOftMsgValueForRelayerRefundLeaf(client, leaf);
+
+    // Currently, msg.value behavior in OFT-supporting Spokes is such that they can't hanlde msg.value being used for
+    // different cases. If both msg value contributions are above 0, we have a bug. Thow
+    if (lineaMsgValuePortion.gt(0) && oftMsgValuePortion.gt(0)) {
+      throw new Error("Invalid configuration: OFT messenger set on Linea chain for relayer refund leaf execution");
+    }
+    const msgValue = lineaMsgValuePortion.add(oftMsgValuePortion);
+    return msgValue.gt(0) ? msgValue : undefined;
+  }
+
+  private async _getOftMsgValueForRelayerRefundLeaf(
+    client: EVMSpokePoolClient,
+    leaf: RelayerRefundLeaf
+  ): Promise<BigNumber> {
+    // ! Todo
+    // This mapping is here to distinguish between chains that have `oftMessengers` storage variable and those that
+    // require msg.value attached on OFT withdrawals. Currently, Arbitrum_Spoke wouldn't handle the msg.value properly
+    // (no refund) so we don't attach that. After Arbitrum_Spoke (and possibly Universal_Spoke) are upgraded to handle
+    // msg.value, we can drop this mapping and instead use response from `oftMessengers` call to decide whether a spoke
+    // supports withdrawals via OFT
+    const CHAINS_SUPPORTING_MSG_VALUE_ON_OFT_WITHDRAWAL = new Set([137]);
```

Comment:
Nitty but I always prefer the alias over using the numbers themselves.
```suggestion
    const CHAINS_SUPPORTING_MSG_VALUE_ON_OFT_WITHDRAWAL = new Set([CHAIN_IDs.POLYGON]);
```